delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/indexing/IndexingStreamCoordinator.scala

```diff
@@ -0,0 +1,74 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.indexing
+
+import akka.actor.typed.ActorSystem
+import ch.epfl.bluebrain.nexus.delta.kernel.RetryStrategy
+import ch.epfl.bluebrain.nexus.delta.kernel.utils.ClassUtils.simpleName
+import ch.epfl.bluebrain.nexus.delta.sdk.indexing.IndexingCommand.{StartIndexing, StopIndexing}
+import ch.epfl.bluebrain.nexus.delta.sdk.indexing.IndexingState._
+import ch.epfl.bluebrain.nexus.delta.sdk.indexing.IndexingStreamCoordinator.Agg
+import ch.epfl.bluebrain.nexus.delta.sdk.syntax._
+import ch.epfl.bluebrain.nexus.sourcing.processor.StopStrategy.TransientStopStrategy
+import ch.epfl.bluebrain.nexus.sourcing.processor.{EventSourceProcessorConfig, ShardedAggregate}
+import ch.epfl.bluebrain.nexus.sourcing.projections.stream.StreamSupervisor
+import ch.epfl.bluebrain.nexus.sourcing.projections.{Projection, ProjectionProgress}
+import ch.epfl.bluebrain.nexus.sourcing.{Aggregate, TransientEventDefinition}
+import fs2.Stream
+import monix.bio.Task
+import monix.execution.Scheduler
+
+import scala.reflect.ClassTag
+
+class IndexingStreamCoordinator[V: ViewLens] private (aggregate: Agg) {
+
+  private def shardedId(view: V) = s"${view.uuid}"
+
+  /**
+    * Start indexing the passed ''view''
+    */
+  def start(view: V): Task[Unit] =
+    aggregate.evaluate(shardedId(view), StartIndexing(view)).mapError(_.value) >> Task.unit
+
+  /**
+    * Stop indexing the passed ''view''
+    */
+  def stop(view: V): Task[Unit] =
+    aggregate.evaluate(shardedId(view), StopIndexing).mapError(_.value) >> Task.unit
+
+}
+
+object IndexingStreamCoordinator {
+
+  type BuildStream[V] = (V, ProjectionProgress[Unit]) => Task[Stream[Task, Unit]]
+
+  private[indexing] type Agg = Aggregate[String, IndexingState, IndexingCommand, IndexingState, Throwable]
+
+  /**
+    * Construct a [[IndexingStreamCoordinator]] relying on the underlying transient [[ShardedAggregate]]
+    */
+  def apply[V: ViewLens](
```

Comment:
If we need extra work to use cluster sharding, we don't want to use it. The balancing benefits do not outperform the extra hussle.

Reply:
all cluster singletons run on the oldest node; this affects the performance of the entire system... but let's discuss in #2031 if you're not confident this would be a good change.